db/crud.go

```diff
@@ -1768,16 +1749,34 @@ func (db *DatabaseCollectionWithUser) SyncFnDryrun(ctx context.Context, body Bod
 	mutableBody, metaMap, _, err := db.prepareSyncFn(oldDoc, newDoc)
 	if err != nil {
 		base.InfofCtx(ctx, base.KeyDiagnostic, "Failed to prepare to run sync function: %v", err)
-		return nil, err, nil
+		return nil, err
 	}
 
 	syncOptions, err := MakeUserCtx(db.user, db.ScopeName, db.Name)
 	if err != nil {
-		return nil, err, nil
+		return nil, err
+	}
+	var output *channels.ChannelMapperOutput
+	var syncErr error
+	if syncFn == "" {
+		output, err = db.ChannelMapper.MapToChannelsAndAccess(ctx, mutableBody, string(oldDoc._rawBody), metaMap, syncOptions)
+		if err != nil {
+			syncErr = fmt.Errorf("%s%s", base.ErrSyncFnDryRun, err)
```

Comment:
I'd be tempted to use `%w` instead of `%s` here for `err` so the original can still be used by `errors.Is`

```suggestion
			syncErr = fmt.Errorf("%s %w", base.ErrSyncFnDryRun, err)
```